master/internal/config/config.go

```diff
@@ -211,3 +211,116 @@ type InternalConfig struct {
 	ExternalSessions  model.ExternalSessions `json:"external_sessions"`
 	PrometheusEnabled bool                   `json:"prometheus_enabled"`
 }
+
+func readPreemptionFromScheduler(conf *resourcemanagers.SchedulerConfig) *bool {
+	if conf == nil || conf.Priority == nil {
+		return nil
+	}
+	return &conf.Priority.Preemption
+}
+
+func readPriorityFromScheduler(conf *resourcemanagers.SchedulerConfig) *int {
+	if conf == nil || conf.Priority == nil {
+		return nil
+	}
+	return conf.Priority.DefaultPriority
+}
+
+// ReadPreemptionStatus resolves the desired preemption status for a job.
+func ReadPreemptionStatus(config *Config, rpName string, jobConf interface{}) bool {
+	if config == nil {
+		panic("input config ptr is null")
+	}
+	var jobPreemptible bool
+	switch jobConf.(type) {
+	case *expconf.ExperimentConfig:
+		jobPreemptible = true
+	case *model.CommandConfig:
+		jobPreemptible = false
+	default:
+		panic("unexpected jobConf type")
+	}
+
+	RMPremption := false // Whether the RM supports preemption
+
+	for _, rpConfig := range config.ResourcePools {
+		if rpConfig.PoolName != rpName {
+			continue
+		}
+		if preemption := readPreemptionFromScheduler(rpConfig.Scheduler); preemption != nil {
+			RMPremption = *preemption
+			return jobPreemptible && RMPremption
+		}
+		break
+	}
+
+	// if not found, fall back to resource manager config
+	if config.ResourceManager.AgentRM != nil {
+		if preemption := readPreemptionFromScheduler(
+			config.ResourceManager.AgentRM.Scheduler,
+		); preemption != nil {
+			RMPremption = *preemption
+			return jobPreemptible && RMPremption
+		}
+	}
+
+	// TODO k8 RM?
+	return jobPreemptible && RMPremption
+}
+
+// ReadPriority resolves the priority value for a job.
+func ReadPriority(config *Config, rpName string, jobConf interface{}) int {
+	if config == nil {
+		panic("input config ptr is null")
+	}
+	var prio *int
+	// look at the idividual job config
+	switch conf := jobConf.(type) {
+	case *expconf.ExperimentConfig:
+		prio = conf.Resources().Priority()
+	case *model.CommandConfig:
+		prio = conf.Resources.Priority
+	}
+	if prio != nil {
+		return *prio
+	}
+
+	var schedulerConf *resourcemanagers.SchedulerConfig
+
+	// if not found, fall back to the resource pools config
+	for _, rpConfig := range config.ResourcePools {
+		if rpConfig.PoolName != rpName {
+			continue
+		}
+		schedulerConf = rpConfig.Scheduler
+	}
+	prio = readPriorityFromScheduler(schedulerConf)
+	if prio != nil {
+		return *prio
+	}
+
+	// if not found, fall back to resource manager config
+	schedulerConf = config.ResourceManager.AgentRM.Scheduler
+	prio = readPriorityFromScheduler(schedulerConf)
+	if prio != nil {
+		return *prio
+	}
+
+	// TODO k8 RM?
+	return resourcemanagers.DefaultSchedulingPriority
```

Comment:
You should be able to read the priority from the kubernetes resource manager - the code will be a little different from reading it from the scheduler, but the priority should be up to date and stored in the groups. Default kubernetes priority is 50, though.

Reply:
> but the priority should be up to date and stored in the groups

we should talk about this part and whether we want it to bubble up to the config level to persist. this is currently what happens with adjusting priority I beleive.